src/hdrgen.d

```diff
@@ -1742,10 +1742,11 @@ public:
         //printf("FuncDeclaration::toCBuffer() '%s'\n", f->toChars());
         if (stcToBuffer(buf, f.storage_class))
             buf.writeByte(' ');
-        typeToBuffer(f.type, f.ident);
+        auto tf = cast(TypeFunction)f.type;
+        typeToBuffer(tf, f.ident);
         if (hgs.hdrgen == 1)
         {
-            if (f.storage_class & STCauto)
+            if (!tf.next)
```

Comment:
Since @dhasenan seems to be working on something else, I took the liberty of making another PR which addresses @UplinkCoder  's feedback and takes a step further solving the destructor header generation. Please review #6423